Improve select radio grid layout (#20034)

SP12893678 opened the pull request:
## Scope

What's changed:

- Use `string-width` package to get more accurate length for some language text.
- Choose minimal size as needed first in current choices length, it will fill entire row at least.

In #20034 flow case.
Chinese lang will use grid-2, becasue the choices length just 2(not grid-3 or grid-4)

![image](https://github.com/user-attachments/assets/97ea9c02-c9a6-4417-bc02-399417e1b24a)


## Potential Risks / Drawbacks

- I am not sure the logic could fit all requirement. Might has some except case?

## Review Notes / Questions

- N/A

---

Fixes #20034

SP12893678 committed:
Improve select radio grid layout (#20034)

rijkvanzanten commented on app/package.json:
```diff
@@ -150,5 +150,8 @@
 		"wellknown": "0.5.0"
+	},
+	"dependencies": {
+		"string-width": "7.2.0"
```
Lets move this to `devDependencies`.

The reason is that we bundle the app before it's published to `npm`, so when you `npm install @directus/app` you already get the full bundled dist, so it shouldn't re-download the dependencies as well 🙂

SP12893678 replied:
Got it. 👍

rijkvanzanten commented on app/src/interfaces/select-radio/select-radio.vue:
```diff
@@ -38,19 +39,25 @@ const gridClass = computed(() => {
 
 	const widestOptionLength = choices.value.reduce((acc, val) => {
-		if (val.text.length > acc.length) acc = val.text;
+		if (stringWidth(val.text) > acc) acc = stringWidth(val.text);
```
I think we do a similar piece of logic in the checkboxes interface as well. Mind checking that? Would be nice if we can squash to birds with one stone 🙂

hanneskuettner replied:
Also can we store it in a variable so we don't have to run it twice?

SP12893678 replied:
Thanks for suggestions!
I have add a util function and test case.

SP12893678 committed:
code review

SP12893678 committed:
fix format

paescuj committed:
Remove unnecessary optional chaining / non-null assertions

paescuj committed:
Remove irrelevant 'value' prop from type

paescuj committed:
Clean-ups

- Return null if choices are empty ([]) since there's no "grid-0" class
- Extend tests

paescuj committed:
Add changeset

paescuj reviewed (approved):
Nice work @SP12893678!
